### PR TITLE
FEM: ResultsPurge - delete all results objects including pipelines and filters

### LINKED ISSUE #13889: FEM ResultsPurge should delete the results pipelines and filters too
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

The FEM ResultsPurge tool should be able to delete not only the CCX Results objects and results meshes but also the pipelines and filters to clear all the results in an Analysis container. At least optionally (e.g. after confirming that the pipelines should be removed as well)

### Full version info

```shell
OS: Windows 10 build 19045
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.37213 (Git)
Build type: Release
Branch: main
```

### Subproject(s) affected?

FEM

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by FEA-eng**
fixes #13889 - quite annoying issue where one cannot easily delete all the results because pipelines and their filters are not affected by the ResultsPurge tool

@marioalexis84 FYI

### CODE REVIEW

**Review by maxwxyz — APPROVED** [review]

File: src/Mod/Fem/femresult/resulttools.py [hunk 1]
```diff
@@ -68,6 +68,16 @@ def purge_results(analysis):
             analysis.Document.removeObject(m.Name)
     analysis.Document.recompute()
```
Comment (marioalexis84) [inline src/Mod/Fem/femresult/resulttools.py:69]: This should also be modified to remove reports from other solvers besides Calculix

**Review by marioalexis84 — CHANGES_REQUESTED** [review]

File: src/Mod/Fem/femresult/resulttools.py (hunk 1 above)
Comment (FEA-eng) [reply src/Mod/Fem/femresult/resulttools.py:71]: Ok, done. Only CalculiX and Elmer create such text documents.

Comment (marioalexis84) [reply src/Mod/Fem/femresult/resulttools.py:71]: It is preferable not to use the object names here. If the name is changed or reports from other solvers are added, the purge will fail.
For the moment, it is better to remove all `App::TextDocument` objects.

Comment (FEA-eng) [reply src/Mod/Fem/femresult/resulttools.py:69]: Ok, PR updated.

**Review by marioalexis84 — APPROVED** [review]

**Comment by marioalexis84** [comment]
@maxwxyz  @chennes please squash commits when merging them.

**Pull request merged**